Makefile

```diff
@@ -1,67 +1,86 @@
+	${MAKE} arch ARCH=armv7k PLATFORM=watchOS HOST=arm-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=arm64_32 PLATFORM=watchOS HOST=arm-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=i386 PLATFORM=watchSimulator HOST=i386-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=x86_64 PLATFORM=watchSimulator HOST=x86_64-apple-darwin MIN_VERSION=watchos-version-min=3.0
+	${MAKE} arch ARCH=armv7 PLATFORM=iPhoneOS HOST=arm-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+	${MAKE} arch ARCH=armv7s PLATFORM=iPhoneOS HOST=arm-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+	${MAKE} arch ARCH=arm64 PLATFORM=iPhoneOS HOST=arm-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+	${MAKE} arch ARCH=arm64e PLATFORM=iPhoneOS HOST=arm-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+	${MAKE} arch ARCH=i386 PLATFORM=iPhoneSimulator HOST=i386-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+	${MAKE} arch ARCH=x86_64 PLATFORM=iPhoneSimulator HOST=x86_64-apple-darwin MIN_VERSION=iphoneos-version-min=7.0
+
+PREFIX = ${CURDIR}/build/${PLATFORM}/${ARCH}
 LIBDIR = ${PREFIX}/lib
 BINDIR = ${PREFIX}/bin
 INCLUDEDIR = ${PREFIX}/include
 
 CXX = ${XCODE_DEVELOPER}/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
 CC = ${XCODE_DEVELOPER}/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-CFLAGS = -isysroot ${IOS_SDK} -I${IOS_SDK}/usr/include -arch ${ARCH} -I${INCLUDEDIR} -miphoneos-version-min=7.0 -O3 -fembed-bitcode
-CXXFLAGS = -stdlib=libc++ -std=c++11 -isysroot ${IOS_SDK} -I${IOS_SDK}/usr/include -arch ${ARCH} -I${INCLUDEDIR} -miphoneos-version-min=7.0 -O3 -fembed-bitcode
-LDFLAGS = -stdlib=libc++ -isysroot ${IOS_SDK} -L${LIBDIR} -L${IOS_SDK}/usr/lib -arch ${ARCH} -miphoneos-version-min=7.0
+CFLAGS = -isysroot ${SDK} -I${SDK}/usr/include -arch ${ARCH} -I${INCLUDEDIR} -m${MIN_VERSION} -O3 -fembed-bitcode
+CXXFLAGS = -stdlib=libc++ -std=c++11 -isysroot ${SDK} -I${SDK}/usr/include -arch ${ARCH} -I${INCLUDEDIR} -m${MIN_VERSION} -O3 -fembed-bitcode
+LDFLAGS = -stdlib=libc++ -isysroot ${SDK} -L${LIBDIR} -L${SDK}/usr/lib -arch ${ARCH} -m${MIN_VERSION}
 
 arch: ${LIBDIR}/libspatialite.a
 
-${LIBDIR}/libspatialite.a: ${LIBDIR}/libproj.a ${LIBDIR}/libgeos.a ${CURDIR}/spatialite
+${LIBDIR}/libspatialite.a: ${LIBDIR}/libproj.a ${LIBDIR}/libgeos.a ${LIBDIR}/libsqlite3.a ${CURDIR}/spatialite
 	cd spatialite && env \
 	CXX=${CXX} \
 	CC=${CC} \
 	CFLAGS="${CFLAGS} -Wno-error=implicit-function-declaration" \
 	CXXFLAGS="${CXXFLAGS} -Wno-error=implicit-function-declaration" \
-	LDFLAGS="${LDFLAGS} -liconv -lgeos -lgeos_c -lc++" ./configure --host=${HOST} --enable-freexl=no --enable-libxml2=no --prefix=${PREFIX} --with-geosconfig=${BINDIR}/geos-config --disable-shared && make clean install-strip
+	LDFLAGS="${LDFLAGS} -liconv -lgeos -lgeos_c -lc++" ./configure --host=${HOST} --enable-freexl=no --enable-libxml2=no --prefix=${PREFIX} --with-geosconfig=${BINDIR}/geos-config --disable-shared && make -j1 clean install-strip
```

Comment:
What's `-j1`? :D

Reply:
Well, passing `-j <number>` to `make` allows to build targets in parallel (`number` should be around 1.5x CPU Cores). However, it seems like `libspatialite` did not properly setup their `Makefile` such that building it with a value create than 1 fails.

=> Just my little hack to allow me to build the dependencies in parallel while building `libspatialite` sequentially.